### PR TITLE
.github: add ruff linter

### PR DESCRIPTION
**Pull request by evgeniiz321**
A workflow to autoverify formatting and some common python style/logic issues of incoming PRs. Ruff is a very fast linter/formatter. Will help us to better check incoming PRs.

### CODE REVIEW

File: neofs-testlib/neofs_testlib_tests/pytest/test_env.py [hunk 1]
```diff
@@ -205,7 +206,7 @@ def test_node_metabase_resync(neofs_env: NeoFSEnv, wallet: NodeWallet, zero_fee)
     for node in neofs_env.storage_nodes:
         node.set_metabase_resync(True)
         node.set_metabase_resync(False)
-    test_http_gw_put_get(neofs_env, wallet, zero_fee)
+    test_gateways_put_get(neofs_env, wallet, zero_fee, "HTTP")
```
Comment (roman-khimov) [inline neofs-testlib/neofs_testlib_tests/pytest/test_env.py:209]: This doesn't look like a linter/formatter change.

File: pytest_tests/lib/helpers/aws_cli_client.py [hunk 2]
```diff
@@ -380,20 +361,7 @@ def delete_bucket_tagging(self, Bucket: str) -> dict:
         output = _cmd_run(cmd)
         return self._to_json(output)
 
-    def put_object_retention(
```
Comment (roman-khimov) [inline pytest_tests/lib/helpers/aws_cli_client.py:383]: This one as well.

**Review by roman-khimov — COMMENTED** [review]
Why?

File: neofs-testlib/neofs_testlib_tests/pytest/test_env.py (hunk 1 above)
Comment (evgeniiz321) [reply neofs-testlib/neofs_testlib_tests/pytest/test_env.py:209]: It actually was shown by the linter. Original method signature was changed sometimes ago, but the corresponding usages were not. Linter showed that there is no test_http_gw_put_get method.

Comment (evgeniiz321) [reply neofs-testlib/neofs_testlib_tests/pytest/test_env.py:209]: But I will move it to a separate commit.

File: pytest_tests/lib/helpers/aws_cli_client.py (hunk 2 above)
Comment (evgeniiz321) [reply pytest_tests/lib/helpers/aws_cli_client.py:383]: This one was also shown by the linter, there were two methods with the same signatures. I will move it to a separate commit.

**Comment by evgeniiz321** [comment]
We don't have a way to enforce formatting and catch some obvious python issues in PRs right now. I wanted to add a workflow with a mix of black/flake8 - but ruff is a much butter choice + I updated some formatting variables, like the length of line.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**